File: 0053/main.go
package main

func maxSubArray(nums []int) int {
	preValue := nums[0]
	maxValue := preValue

	for _, value := range nums[1:] {
		if preValue > 0 {
			preValue += value
		} else {
			preValue = value
		}

		if preValue > maxValue {
			maxValue = preValue
		}
	}
	return maxValue
}

File: 0053/main.swift
class Solution {
    func maxSubArray(_ nums: [Int]) -> Int {
        var preValue = nums[0]
        var maxValue = preValue
        for value in nums.dropFirst(){
            if preValue > 0{
                preValue += value
            } else {
                preValue = value
            }
            if preValue > maxValue{
                maxValue = preValue
            }
        }
        return maxValue
    }
}